src/components/editor/metaform-editor-right-drawer-visibility.tsx

```diff
@@ -0,0 +1,338 @@
+import { Divider, FormControl, FormControlLabel, InputLabel, MenuItem, Select, Switch, Typography } from "@mui/material";
+import { Metaform, MetaformField, MetaformSection } from "generated/client";
+import produce from "immer";
+import strings from "localization/strings";
+import React, { useEffect } from "react";
+
+/**
+ * Component properties
+ */
+interface Props {
+  sectionIndex?: number;
+  fieldIndex?: number;
+  pendingForm: Metaform;
+  setPendingForm: (metaform: Metaform) => void;
+}
+
+/**
+ * Draft editor right drawer component
+ */
+const MetaformEditorRightDrawerVisibleCondition: React.FC<Props> = ({
+  sectionIndex,
+  fieldIndex,
+  pendingForm,
+  setPendingForm
+}) => {
+  const [ disabledValue, setDisabledValue ] = React.useState<boolean>(false);
+  const [ requiredConditionField, setRequiredConditionField ] = React.useState<string>("");
+  const [ requiredConditionEqualsValue, setRequiredConditionEqualsValue ] = React.useState<string>("");
+  const [ requiredConditionInfoField, setRequiredConditionInfoField ] = React.useState<boolean>(true);
+  const [ selectedComponent, setSelectedComponent] = React.useState<string | undefined>("");
+  const [ metaFormFieldSection, setMetaFormFieldSection] = React.useState<MetaformField | MetaformSection>();
+
+  /**
+   * Set values of Confidition field, switch and get selected component name
+   * 
+   */
+  const setVisiblityComponentValues = () => {
+    if (fieldIndex === undefined && sectionIndex !== undefined) {
+      const section = pendingForm.sections![sectionIndex].visibleIf;
+      setMetaFormFieldSection(pendingForm.sections![sectionIndex]);
+      setDisabledValue(section !== undefined);
+      setRequiredConditionField(section ? section.field! : "");
+      setRequiredConditionEqualsValue(section ? section.equals! : "");
+      setRequiredConditionInfoField(!section);
+      setSelectedComponent(pendingForm.sections![sectionIndex].title);
+    }
+
+    if (fieldIndex !== undefined && sectionIndex !== undefined) {
+      const field = pendingForm.sections![sectionIndex].fields![fieldIndex].visibleIf;
+      setMetaFormFieldSection(pendingForm.sections![sectionIndex].fields![fieldIndex]);
+      setSelectedComponent(pendingForm.sections![sectionIndex].fields![fieldIndex].name);
+      setDisabledValue(field !== undefined);
+      setRequiredConditionField(field ? field.field! : "");
```

Comment:
Would it work if you'd check if _field_ exists, store that into a variable and therefore you wouldn't need to use ternary inside these setters?
```suggestion
      const visibleIfField = field?.field
      setRequiredConditionField(field ?? : "");
```
Propably do not apply this suggestion, but use it a reference instead.